static/app/views/dashboards/widgets/timeSeriesWidget/plottables/plottable.tsx

```diff
@@ -6,11 +6,16 @@ import type {
   RateUnit,
   SizeUnit,
 } from 'sentry/utils/discover/fields';
+import type {TimeSeries} from 'sentry/views/dashboards/widgets/common/types';
 
 /**
  * A `Plottable` is any object that can be converted to an ECharts `Series` and therefore plotted on an ECharts chart. This could be a data series, releases, samples, and other kinds of markers. `TimeSeriesWidgetVisualization` uses `Plottable` objects under the hood, to convert data coming into the component via props into ECharts series.
  */
 export interface Plottable {
+  /**
+   * Allows the user to clone the plottable w/ an option to mutate the TimeSeries
+   */
+  clone(cb?: (timeSeries: TimeSeries) => TimeSeries): this;
```

Comment:
Unfortunately, this won't work nicely very soon. Other plottables (e.g., "Samples") don't have a backing `TimeSeries`, they're backed by another data structure. Any methods on the `Plottable` interface need to be ignorant of the backing data.

What do you think about having a method like `.slice` (or maybe `.zoom`, or `.constrain`) that returns a clone of the `Plottable` but with the sliced data set? Every implementation of `Plottable` would then implement the method based on its own underlying data. I think that's also a little tidier than the callback approach

Reply:
I guess `.slice` is awkward because it's possible to accidentally _expand_ the plottable, which is bad. The important part is that we need a way to make this method independent from the underlying data somehow

Reply:
So something like this?

```suggestion
  /**
   * Returns a cloned Plottable, constraining any time-series data within the date boundaries
   */
  constrain(boundaryStart: Date, boundaryEnd: Date): this;
```

Reply:
That's what I'm imagining, yes!